tests/e2e/e2e_ica_test.go

```diff
@@ -0,0 +1,168 @@
+package e2e
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// TestICARegister must run before any other
+func (s *IntegrationTestSuite) TestICA_1_Register() {
+	s.Run("register_ICA", func() {
+		connectionID := "connection-0"
+		var owner string
+		ownerAddr, err := s.chainA.genesisAccounts[1].keyInfo.GetAddress()
+		s.Require().NoError(err)
+		owner = ownerAddr.String()
+		s.registerICA(owner, connectionID)
+
+		time.Sleep(2 * time.Minute)
+
+		chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+		s.Require().Eventually(
+			func() bool {
+				icaAddr, err := queryICAaddr(chainAAPIEndpoint, owner, connectionID)
+				s.T().Logf("%s's interchain account on chain %s: %s", owner, s.chainB.id, icaAddr)
+				s.Require().NoError(err)
+				return owner != "" && icaAddr != ""
+			},
+			2*time.Minute,
+			10*time.Second,
+		)
+	})
+}
+
+func (s *IntegrationTestSuite) TestICA_2_BankSend() {
+	s.Run("test ica transactions", func() {
+		chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+		chainBAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainB.id][0].GetHostPort("1317/tcp"))
+		connectionID := "connection-0"
+		// step 1: get ica addr
+		icaOwnerAddr, err := s.chainA.genesisAccounts[1].keyInfo.GetAddress()
```

Comment:
this account is used in multiple places, it would be good to make a variable out of it.
since it's not possible to save as a global due to the need to wait until the chain is initialized, at a minimum you could use something like:
```
var icaOwnerAccountIndex = 1
...
icaOwnerAddr, err := s.chainA.genesisAccounts[icaOwnerAccountIndex].keyInfo.GetAddress()
```
and then ensure the index is the same everytime you need to ask for the `icaOwnerAddr`